Sources/Apollo/ApolloClientProtocol.swift

```diff
@@ -82,4 +82,89 @@ public protocol ApolloClientProtocol: AnyObject {
+  ///
+  /// - Parameters:
+  ///   - query: The query to fetch.
+  ///   - cachePolicy: A cache policy that specifies when results should be fetched from the server and when data should be loaded from the local cache.
+  ///   - queue: A dispatch queue on which the result handler will be called. Should default to the main queue.
+  ///   - contextIdentifier: [optional] A unique identifier for this request, to help with deduping cache hits for watchers. Should default to `nil`.
+  ///   - resultHandler: [optional] A closure that is called when query results are available or when an error occurs.
+  /// - Returns: An object that can be used to cancel an in progress fetch.
+  @available(macOS 10.15, iOS 13, watchOS 8, tvOS 15, *)
+  func fetch<Query: GraphQLQuery>(query: Query,
+                                  cachePolicy: CachePolicy,
+                                  contextIdentifier: UUID?,
+                                  queue: DispatchQueue) async throws -> GraphQLResult<Query.Data>
+
+  /// Performs a mutation by sending it to the server.
+  ///
+  /// - Parameters:
+  ///   - mutation: The mutation to perform.
+  ///   - publishResultToStore: If `true`, this will publish the result returned from the operation to the cache store. Default is `true`.
+  ///   - queue: A dispatch queue on which the result handler will be called. Should default to the main queue.
+  ///   - resultHandler: An optional closure that is called when mutation results are available or when an error occurs.
+  /// - Returns: An object that can be used to cancel an in progress mutation.
+  @available(macOS 10.15, iOS 13, watchOS 8, tvOS 15, *)
+  func perform<Mutation: GraphQLMutation>(mutation: Mutation,
+                                          publishResultToStore: Bool,
+                                          queue: DispatchQueue) async throws -> GraphQLResult<Mutation.Data>
+
+  /// Uploads the given files with the given operation.
+  ///
+  /// - Parameters:
+  ///   - operation: The operation to send
+  ///   - files: An array of `GraphQLFile` objects to send.
+  ///   - queue: A dispatch queue on which the result handler will be called. Should default to the main queue.
+  ///   - completionHandler: The completion handler to execute when the request completes or errors. Note that an error will be returned If your `networkTransport` does not also conform to `UploadingNetworkTransport`.
+  /// - Returns: An object that can be used to cancel an in progress request.
+  @available(macOS 10.15, iOS 13, watchOS 8, tvOS 15, *)
+  func upload<Operation: GraphQLOperation>(operation: Operation,
+                                           files: [GraphQLFile],
+                                           queue: DispatchQueue) async throws -> GraphQLResult<Operation.Data>
+
+#elseif swift(>=5.5.0) && swift(<5.5.2) && canImport(_Concurrency)
```

Comment:
@designatednerd I made the conditional compilation flags more specific, and it looks like `testAsync` is passing on both Xcode 13.0.0, and Xcode 13.2.1. There's a ton of code repetition though. I can see what I can do about that if you decide we should move forward with this PR.

Reply:
[Here's](https://github.com/swift-server/swift-service-lifecycle/issues/110) a discussion on `canImport(_Concurrency)` that was helpful.